crypto/schemes/enc/v1/README.md

```diff
@@ -0,0 +1,146 @@
+# Dapr encryption scheme v1: `dapr.io/enc/v1`
+
+This document contains the reference for the **Dapr encryption scheme v1**, identified by the header `dapr.io/enc/v1`.
+
+The Dapr encryption scheme is optimized for processing data as a stream. Data is chunked into multiple parts which are encrypted independently. This allows us to return data to callers as a stream, even when decrypting messages, being confident that we are not flushing unverified data to the client.
+
+> **Sources:** The encryption scheme that Dapr uses is heavily inspired by the [Tink wire format](https://developers.google.com/tink/wire-format) (from the Tink library maintained by Google), as well as by Filippo Valsorda's [age](https://age-encryption.org/v1), and Minio's [DARE](https://github.com/minio/sio).
+
+## Key
+
+Each message is encrypted with a 256-bit symmetric **File Key (FK)** that is randomly generated by Dapr for each new message. The key must be generated as 32 byte of output from a CSPRNG (such as Go's `crypto/rand.Reader`) and must not be reused for other files.
+
+The FK is wrapped using a key stored in a key vault (**Key Encryption Key (KEK)**) by Dapr. The result of the wrapping operation is the **Wrapped File Key (WFK)**. The algorithm used depends on the type of the KEK as well as the algorithms supported by the component:
+
+- For symmetric keys:
+    - AES-KW with 256-bit keys ([RFC 3394](https://www.rfc-editor.org/rfc/rfc3394.html)): `A256KW`
+		- Because the File Key is 256-bit long, only 256-bit wrapping keys can be used
+	- AES-CBC with keys 128-bit, 192-bit, and 256-bit: `A128CBC-NOPAD`, `A192CBC-NOPAD`, `A256CBC-NOPAD`
+		- These don't use PKCS#7 padding because the File Key is 256-bit so it's a multiple of the AES block size.
+- For RSA keys:
+    - RSA OAEP with SHA-256: `RSA-OAEP-256`
+		- Dapr doesn't impose limitations on the size of the key, and any key bigger than 1024 bits should work; however, 4096-bit keys are strongly recommended.
+
+> Other key wrapping algorithms can be implemented in the future.
+
+## Ciphertext format
+
+The ciphertext is formatted as:
+
+```text
+header || binary_payload
+```
+
+## Header
+
+The **header** is human-readable and contains 3 items, each terminated by a line feed (`0x0A`) character:
+
+1. Name and version of the encryption scheme used. In this version of the spec, this is always `dapr.io/enc/v1`.
+2. The manifest, which is a JSON object, compacted with all the unnecessary whitespaces removed. 
```

Comment:
Might be interesting to link to JSON RFC as it defines what __unnecessary whitespace__. I know that for go this is a non-issue as that seems to be the default for `encoding/json.Marshall`. Ditto for nodejs. But that isn't the default for [python]( https://stackoverflow.com/questions/16311562/python-json-without-whitespaces) AFAIK.